service/history/shardContext.go

```diff
@@ -56,6 +56,7 @@ type (
 		GetMetricsClient() metrics.Client
 		GetTimerAckLevel() time.Time
 		UpdateTimerAckLevel(ackLevel time.Time) error
+		Close()
```

Comment:
All the life-cycle of shard is managed within shardController.  Do you think that instead of exposing the Close method we just Close ExecutionManager within ShardController?